evil-collection.el

```diff
@@ -112,6 +114,62 @@ this confusing. It will be included if
   :type '(repeat (choice symbol sexp))
   :group 'evil-collection)
 
+;;;###autoload
+(cl-defun evil-collection-translate-key (state keymap-name
+                                               &rest maps
+                                               &key destructive
+                                               &allow-other-keys)
+  "Translate keys in the keymap corresponding to STATE and KEYMAP-NAME.
+STATE should be the name of an evil state or nil. KEYMAP-NAME should be a symbol
+corresponding to the keymap to make the translations in. MAPS corresponds to a
+list of translations (key replacement pairs). For example, specifying \"a\"
+\"b\" will bind \"a\" to \"b\"'s definition in the keymap.If DESTRUCTIVE is
+non-nil, the keymap will be destructively altered without a backup being
+created. If DESTRUCTIVE is nil, a backup of the keymap will be stored on the
+initial invocation, and future invocations will always look up keys in the
+backup keymap. On the other hand, if DESTRUCTIVE is non-nil, calling this
+function multiple times with \"a\" \"b\" \"b\" \"a\", for example, would
+continue to swap and unswap the definitions of these keys. This means that when
+DESTRUCTIVE is non-nil, all related swaps/cycles should be done in the same
+invocation."
+  (declare (indent defun))
+  (let* ((keymap (if state
+                     (evil-get-auxiliary-keymap
+                      (symbol-value keymap-name) state t t)
+                   (symbol-value keymap-name)))
+         (backup-keymap-name (intern (format
+                                      "evil-collection-%S-%S-state-backup-map"
+                                      keymap-name
+                                      (or state 'no))))
+         (lookup-keymap (if (and (not destructive)
```

Comment:
I see we're playing around with destructive.

When would you recommend one or the other? (e.g. swapping keys with destructive set to nil and vice versa)